src/main/java/Duke.java

```diff
@@ -1,10 +1,186 @@
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    //taskList variable for level-2 onwards
+    private static Task[] taskList = new Task[100];
+    //task count
+    private static int taskCount = 0;
+
+
+    //echo function for level-1
+    public static void echo(String message) {
+        System.out.println(message);
+    }
+
+
+    //add function for level-2
+    public static void add(String task) {
+        Task newTask = new Task(task);
+        taskList[taskCount] = newTask;
+        taskCount++;
+        System.out.println("added: " + task);
+    }
+
+
+    //list function for level-2
+    public static void list() {
+        for (int i = 0; i < taskCount; i++) {
+            System.out.print((i + 1) + ". ");
+            System.out.println(taskList[i].toString());
+        }
+    }
+
+
+    //mark function for level-3
+    public void mark(int index) {
+        if (index > taskCount || index < 0) {
+            System.out.println("Index out of bounds");
+            return;
+        }
+        System.out.println("Nice! I've marked this task as done:");
+        taskList[index - 1].markAsDone();
+    }
+
+
+    //unmark function for level-3
+    public void unmark(int index) {
+        if (index > taskCount || index < 0) {
+            System.out.println("Index out of bounds");
+            return;
+        }
+        System.out.println("Ok, I've marked this task as not done yet:");
+        taskList[index - 1].markAsNotDone();
+
+    }
+
+
+    //add toDo task for level-4
+    public static void addToDo(String[] inputComponent) {
```

Comment:
Perhaps use the plural form like inputComponents? As it is representing a collection of objects